### PR TITLE
Add Provider

### PR DESCRIPTION
**Pull request by itdependsnetworks**
Added provider similar to that used in other networking modules. 

Note the following: 
- Local params are preferred over provider defined params
- Can use either hostname or host, I added host since that is used in other network modules, but hostname is preferred. 
- Will do the same for napalm_install_config once this PR is deemed good. 

pinging @jedelman8

### CODE REVIEW

File: library/napalm_get_facts.py [hunk 1]
```diff
@@ -129,6 +131,17 @@ def main():
     if not napalm_found:
         module.fail_json(msg="the python module napalm is required")
 
+    provider = module.params['provider'] or {}
+
+    # allow host or hostname
+    if provider.get('host') and not provider.get('hostname'):
+        provider['hostname'] = provider.get('host')
+    # allow local params to override provider
```
Comment (jedelman8) [inline library/napalm_get_facts.py:163]: If supporting `host` too, should add as an alias in the `argument_spec`

**Comment by dbarrosop** [comment]
@itdependsnetworks good stuff : ) Could you add an example on how to use it in the docstring?

Thanks!

**Comment by itdependsnetworks** [comment]
Updated doc string

**Comment by dbarrosop** [comment]
Great! Thx @itdependsnetworks

**Pull request merged**